core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -4866,54 +4855,37 @@ private void multiClaimQueryBuilder(SqlBuilder sqlBuilder, SqlBuilder header, bo
     }
 
     private void buildClaimWhereConditions(SqlBuilder sqlBuilder, String attributeName, String operation,
-            String attributeValue) {
-
-        sqlBuilder.where("UA.UM_ATTR_NAME = ?", attributeName);
-
-        if (ExpressionOperation.EQ.toString().equals(operation)) {
-            if (isCaseSensitiveUsername()) {
-                sqlBuilder.where("UA.UM_ATTR_VALUE = ?", attributeValue);
-            } else {
-                sqlBuilder.where("LOWER(UA.UM_ATTR_VALUE) = LOWER(?)", attributeValue);
-            }
-        } else if (ExpressionOperation.EW.toString().equals(operation)) {
-            if (isCaseSensitiveUsername()) {
-                sqlBuilder.where("UA.UM_ATTR_VALUE LIKE ?", "%" + attributeValue);
-            } else {
-                sqlBuilder.where("LOWER(UA.UM_ATTR_VALUE) LIKE LOWER(?)", "%" + attributeValue);
-            }
-        } else if (ExpressionOperation.CO.toString().equals(operation)) {
+                                           String attributeValue) {
+
+        if (ExpressionOperation.NE.toString().equals(operation)) {
+            /*
+             * When operation is NE (Not Equal), consider both users whose attribute value does not match the
+             * specified value and users who do not have the attribute configured.
+             */
+
+            // Build a subquery to identify users to exclude.
+            SqlBuilder usersToExcludeSqlBuilder = new SqlBuilder(
```

Comment:
Add to a separate private method

Reply:
The `ne` operator is checked first because it does not require appending `sqlBuilder.where("UA.UM_ATTR_NAME = ?", attributeName);`. Placing the `ne` condition at the beginning helps avoid this clause from being added unnecessarily.